src/context/TransactionContext.jsx

```diff
@@ -29,9 +29,8 @@ export const TransactionsProvider = ({ children }) => {
     const getAllTransactions = async () => {
         try {
             if (ethereum) {
-                const transactionsContract = createEthereumContract();
-
-                const availableTransactions = await transactionsContract.getAllTransactions();
+                const contract = createEthereumContract();
```

Comment:
```suggestion
```

It is recommended not to create a contract instance inside a method, as this would cause the contract instance to be recreated every time the function is called, when in fact we only need to do this once.

(The same situation exists in other functions.)